### PR TITLE
Fix KernelTests#notLinux test

### PR DESCRIPTION
**Pull request by parttimenerd**
### Description

Fix the timeout of the `KernelTests#notLinux` test by not running it with `/usr/bin/sudo`.

### Motivation and context

Running `/usr/bin/sudo` on Mac as a non-root user, asks the user for a password infinitely, so the running the profiler with sudo causes timeouts on Mac. 

### How has this been tested?

Ran it on Mac with different Java versions.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: test/test/kernel/KernelTests.java [hunk 1]
```diff
@@ -38,7 +38,7 @@ public void fdtransfer(TestProcess p) throws Exception {
     @Test(mainClass = ListFiles.class, jvmArgs = "-XX:+UseParallelGC -Xmx1g -Xms1g", os = {Os.MACOS, Os.WINDOWS})
     public void notLinux(TestProcess p) throws Exception {
         try {
-            p.profile("-e cpu -d 3 -i 1ms -o collapsed -f %f --fdtransfer", true);
+            p.profile("-e cpu -d 3 -i 1ms -o collapsed -f %f --fdtransfer", false);
```
Comment (apangin) [inline test/test/kernel/KernelTests.java:41]: The argument may be removed altogether: sudo=false is the default.

Comment (parttimenerd) [reply test/test/kernel/KernelTests.java]: done

File: test/one/profiler/test/TestProcess.java [hunk 2]
```diff
@@ -248,14 +248,16 @@ public Output profile(String args) throws IOException, TimeoutException, Interru
...
+	if (sudo) {
+		System.err.println("    sudo");
             cmd.add("/usr/bin/sudo");
         }
         cmd.add("build/bin/asprof");
         addArgs(cmd, args);
         cmd.add(Long.toString(pid()));
         log.log(Level.FINE, "Profiling " + cmd);
-
+System.out.println("   command " + cmd);
```
Comment (apangin) [inline test/one/profiler/test/TestProcess.java:260]: Is it some debugging leftover?
A command is already logged in the above statement.

File: test/test/kernel/KernelTests.java [hunk 3]
```diff
@@ -38,9 +38,10 @@ public void fdtransfer(TestProcess p) throws Exception {
     @Test(mainClass = ListFiles.class, jvmArgs = "-XX:+UseParallelGC -Xmx1g -Xms1g", os = {Os.MACOS, Os.WINDOWS})
     public void notLinux(TestProcess p) throws Exception {
         try {
-            p.profile("-e cpu -d 3 -i 1ms -o collapsed -f %f --fdtransfer", true);
+            p.profile("-e cpu -d 3 -i 1ms -o collapsed -f %f --fdtransfer");
             throw new AssertionError("FdTransferClient should succeed on Linux only");
         } catch (IOException e) {
+            System.err.println("file content: " + p.readFile(TestProcess.PROFERR));
```
Comment (apangin) [inline test/test/kernel/KernelTests.java:44]: Is it also a leftover?
All output files are saved in `build/test/logs`.

**Comment by apangin** [comment]
Merged. Thanks

**Pull request merged**